### DISCUSSION TITLE
Add the method set_raw_body to the response object in the protobuf definition

### DISCUSSION THREAD

**Discussion by dszczyt** [Kong/kong#13777] [Ideas and feature requests]
The set_raw_body function exists on responses for lua plugins, but it [doesn't exist ](https://github.com/Kong/kong/blob/master/kong/include/kong/pluginsocket.proto#L325) for example in the golang pdk, because it's auto generated from the protobuf and its declaration is done only the for request object. Could we add it ? It could allow to make chainable plugins in go. Here we have to use the "exit" function to return the body, but it prevents us to pass the response to another plugin.

**Comment by bungle** [comment]
@dszczyt,  we have `kong.response.set_raw_body` but that only works in `body_filter` phase, The `body_filter` phase is not even available in Go plugins. I believe you are talking either the `access` / `response` phase.
